### ISSUE TITLE
no-unknown-classes reports sibling bindings and object keys of a destructured initializer

### ISSUE THREAD

**Issue by CrysisDeu** [shadcn-ui/lint#38]
## Summary

`no-unknown-classes` treats every member of a destructured array/object initializer as a class when only one binding flows into `className`. It reports the other array elements (a `var(--ok)` used as `style.background`, a plain label like `Connected`) and, for objects, the **property keys** (`Icon`, `colorCls`) rather than values.

Reproduced on `@shadcn/lint` 0.1.0 and 0.1.1, ESLint 9.39.4, `@typescript-eslint/parser` 8.58.0, Tailwind 4.3.x.

## Repro

`src/a.tsx`:

```tsx
export function A({ ok }: { ok: boolean }) {
  const [dot, text, cls] = ok ? ['var(--ok)', 'Connected', 'text-ok'] : ['var(--muted)', 'Needs setup', 'text-muted']
  const { Icon, colorCls } = ok ? { Icon: 'x', colorCls: 'text-ok' } : { Icon: 'y', colorCls: 'text-muted' }
  return <span className={`flex ${cls} ${colorCls}`} style={{ background: dot }} title={Icon}>{text}</span>
}
```

`eslint.config.mjs`:

```js
import { plugin as shadcn } from '@shadcn/lint'
import tsParser from '@typescript-eslint/parser'
export default [{
  files: ['src/**/*.tsx'],
  languageOptions: { parser: tsParser, parserOptions: { ecmaFeatures: { jsx: true } } },
  plugins: { shadcn },
  rules: { 'shadcn/no-unknown-classes': 'error' },
}]
```

`src/index.css`: `@import "tailwindcss"; @theme { --color-ok: #0a0; --color-muted: #888; }`

## Actual

```
2:34  error  "var(--ok)" is not a class this project's Tailwind knows …
2:47  error  "Connected" is not a class …
2:74  error  "var(--muted)" is not a class …
2:90  error  "Needs" is not a class …
2:90  error  "setup" is not a class …
3:37  error  "Icon" is not a class …
3:48  error  "colorCls" is not a class …
3:74  error  "Icon" is not a class …
3:85  error  "colorCls" is not a class …
✖ 9 problems (9 errors, 0 warnings)
```

## Expected

No report: `cls` and `colorCls` resolve to `text-ok` / `text-muted`, which the theme declares. The sibling bindings never reach a class site.

For the array case, only the element at the destructured index that flows into a class site should be read (index 2 here). For the object case, only the **value** of the destructured property (`colorCls`) should be read — the key names are being reported as classes.

## Context

Hit this on a ~1,500-file dashboard while adopting the rule: 28 of the 295 initial `no-unknown-classes` reports were this shape (a `[dotColor, label, textClass]` tuple pattern repeated across five settings panels, plus two `{ Icon, colorCls }` lookups).
